src/OpenLoco/src/Vehicles/Orders.cpp

```diff
@@ -206,6 +206,68 @@ namespace OpenLoco::Vehicles
         return begin();
     }
 
+    uint8_t swapAdjacentOrders(Order& a, Order& b)
+    {
+        const auto rawOrderA = a.getRaw();
+        const auto rawOrderB = b.getRaw();
+        const auto lengthOrderA = kOrderSizes[enumValue(a.getType())];
+        const auto lengthOrderB = kOrderSizes[enumValue(b.getType())];
+
+        // Ensure the two orders are indeed adjacent
+        assert(&a + lengthOrderA == &b);
+
+        // Copy B over A, and append B right after
+        const auto dest = reinterpret_cast<uint8_t*>(&a);
+        std::memcpy(dest, &rawOrderB, lengthOrderB);
+        std::memcpy(dest + lengthOrderB, &rawOrderA, lengthOrderA);
+
+        // Return the length with which to offset
+        return lengthOrderB;
+    }
+
+    uint16_t reverseVehicleOrderTable(uint32_t tableOffset, uint32_t orderOfInterest)
+    {
+        // Retrieve list of raw orders
+        std::vector<uint64_t> rawOrders{};
+        Vehicles::OrderRingView orders(tableOffset);
+        for (auto& order : orders)
+        {
+            rawOrders.push_back(order.getRaw());
+        }
+
+        // Nothing to do? Bail early
+        if (rawOrders.size() == 0)
+        {
+            return orderOfInterest;
+        }
+
+        // Keep track of the type of the order of interest
+        auto ooiType = orders.atIndex(orderOfInterest)->getType();
+
+        // Figure out where the order table starts in memory
+        auto firstOrder = reinterpret_cast<uint8_t*>(orders.atIndex(0));
+        auto dest = firstOrder;
+
+        // Write reversed list over existing list
+        for (auto it = rawOrders.rbegin(); it != rawOrders.rend(); ++it)
+        {
+            auto rawOrder = *it;
+            auto orderType = rawOrder & 0x7;
+            auto orderLength = kOrderSizes[orderType];
+            std::memcpy(dest, &rawOrder, orderLength);
+            dest += orderLength;
+        }
+
+        printf("First order @ %p\n", firstOrder);
+        printf("End of table @ %p\n", dest);
+
+        // Figure out the new position of the order of interest
+        auto newOOIOffset = dest - orderOfInterest - kOrderSizes[enumValue(ooiType)];
+        printf("Calculated OOI @ %p\n", newOOIOffset);
```

Comment:
This does not appear to be yielding the correct offset at present. Not sure what the problem is yet.